src/main/java/net/dries007/tfc/common/capabilities/SnitchingFluidTank.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Licensed under the EUPL, Version 1.2.
+ * You may obtain a copy of the Licence at:
+ * https://joinup.ec.europa.eu/collection/eupl/eupl-text-eupl-12
+ */
+
+package net.dries007.tfc.common.capabilities;
+
+import java.util.function.Predicate;
+
+import net.minecraftforge.fluids.FluidStack;
+import net.minecraftforge.fluids.capability.templates.FluidTank;
+
+public class SnitchingFluidTank extends FluidTank
```

Comment:
```suggestion
public class InventoryFluidTank extends FluidTank
```

Like `InventoryItemHandler` / `ISlotCallback`. This is just the fluid tank impl. that we use in inventories.